### ISSUE TITLE
Distinct steps parallelism

### ISSUE THREAD

**Issue by jkopczyn** [buildkite/docs#1204]
The [documentation about parallel builds](https://buildkite.com/docs/tutorials/parallel-builds) talks a lot about what's needed and what's possible for running a single step several times in parallel. It says nothing at all about what the default behavior is for _different_ steps run in parallel. Some of that is covered in [dependencies](https://buildkite.com/docs/pipelines/dependencies), but not all.

Two main changes I think are needed:

- Link from Parallel Builds to Dependencies clarifying that the parallelism for distinct steps is controlled primarily by dependencies.
- Explicit discussion of how specifying agents and queues affects distinct steps.

Would also be nice to have something about steps which create a resource later steps use (e.g. create a docker container later steps SSH into), but that's not entirely within this topic.

**Comment by plaindocs** [comment]
Hey @jkopczyn thanks for your patience, and thanks for taking the time to help us improve.
We'll push some updates to these docs soon.

**Comment by karensawrey** [comment]
Solved by https://github.com/buildkite/docs/pull/1627
